packages/@react-aria/table/stories/example-backwards-compat.tsx

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {mergeProps} from '@react-aria/utils';
+import React, {ReactNode, useRef, useState} from 'react';
+import {useCheckbox} from '@react-aria/checkbox';
+import {useFocusRing} from '@react-aria/focus';
+import {
+  useTable,
+  useTableCell,
+  useTableColumnHeader,
+  useTableHeaderRow,
+  useTableRow,
+  useTableRowGroup, useTableSelectAllCheckbox,
+  useTableSelectionCheckbox
+} from '../src';
+import {useTableState} from '@react-stately/table';
+import {useToggleState} from '@react-stately/toggle';
+import {VisuallyHidden} from '@react-aria/visually-hidden';
+
+export function Table(props) {
+  let {onAction} = props;
+  let [showSelectionCheckboxes, setShowSelectionCheckboxes] = useState(props.selectionStyle !== 'highlight');
+  let state = useTableState({
+    ...props,
+    showSelectionCheckboxes,
+    selectionBehavior: props.selectionStyle === 'highlight' ? 'replace' : 'toggle'
+  });
+  // If the selection behavior changes in state, we need to update showSelectionCheckboxes here due to the circular dependency...
+  let shouldShowCheckboxes = state.selectionManager.selectionBehavior !== 'replace';
+  if (shouldShowCheckboxes !== showSelectionCheckboxes) {
+    setShowSelectionCheckboxes(shouldShowCheckboxes);
```

Comment:
Good catch adding this here (I presume you added it because our own TableView needed), do we need to add it to the useTable docs example?

Reply:
yeah, i just wanted to make it behave more like ours, it doesn't really change anything, but we could put it in our docs as well if others wanted to do it